### PR TITLE
JENKINS-24697 Add support for a validation query to prevent plugins getting a bad database handle returned

### PR DESCRIPTION
**Pull request by davidvanlaatum**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,16 @@
...
+      <dependency>
+          <groupId>javax.validation</groupId>
+          <artifactId>validation-api</artifactId>
+          <version>1.0.0.GA</version>
+      </dependency>
+      <dependency>
+          <groupId>org.hibernate</groupId>
+          <artifactId>hibernate-validator</artifactId>
+          <version>4.1.0.Final</version>
+      </dependency>
```
Comment (uhafner) [inline pom.xml]: Is there a reason that you are mixing versions?

File: src/main/resources/org/jenkinsci/plugins/database/AbstractRemoteDatabase/config.groovy [hunk 2]
```diff
@@ -14,6 +14,9 @@ f.entry(field:"username", title:_("Username")) {
 f.entry(field:"password", title:_("Password")) {
     f.password()
 }
+f.entry(field:"validationQuery", title:_("Validation Query")) {
+  f.textarea()
+}
 f.advanced {
```
Comment (uhafner) [inline src/main/resources/org/jenkinsci/plugins/database/AbstractRemoteDatabase/config.groovy]: Can you add a help text on how to use this query?

**Comment by uhafner** [comment]
It would simplify the history if you could squash several commits to a single one...

File: pom.xml (hunk 1 above)
Comment (davidvanlaatum) [inline pom.xml]: Something somewhere depended on ver 4.1 as a test dependency but I can't find it now Ill have another look at work tomorrow.

Comment (davidvanlaatum) [inline pom.xml]: hibernate-parent pom for 3.6.10.Final has 
```xml
<dependency>
  <groupId>org.hibernate</groupId>
  <artifactId>hibernate-validator</artifactId>
  <version>4.1.0.Final</version>
</dependency>
```
in dependencyManagement which is pulled in as a test dependency of hibernate-core

File: src/main/resources/org/jenkinsci/plugins/database/AbstractRemoteDatabase/config.groovy (hunk 2 above)
Comment (KostyaSha) [inline src/main/resources/org/jenkinsci/plugins/database/AbstractRemoteDatabase/config.groovy]: Also if it supposed to be a code and if it supported by codemirror, then you can use codemirror-config parameter.
Help page is also required.

File: pom.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.509</version><!-- which version of Jenkins is this plugin built against? -->
+        <version>1.565.2</version><!-- which version of Jenkins is this plugin built against? -->
```
Comment (KostyaSha) [inline pom.xml]: please keep existed indents.
Also please replace this wrong `<!-- which version of Jenkins is this plugin built against? -->` message, with description of why this minimal core version is required.

**Comment by davidvanlaatum** [comment]
Think I have squashed correctly but not sure first time I have done it

**Comment by uhafner** [comment]
Seems that there are still a lot of commits: https://github.com/jenkinsci/database-plugin/pull/1/commits

Normally there should be only one remaining...

**Comment by uhafner** [comment]
And the help page still is missing...

**Comment by KostyaSha** [comment]
@daniel-beck ?

**Comment by uhafner** [comment]
Wouldn't it make sense to address all issues *before* merging?

**Comment by davidvanlaatum** [comment]
Woops thought I had but now I realize I forgot to push ;)

**Comment by daniel-beck** [comment]
@KostyaSha Got the wrong autocompletion for `da`, or why did you summon me?

**Pull request merged**